### PR TITLE
managed app sample

### PR DESCRIPTION
**Pull request by thomasyip-msft**
## Purpose
Managed app sample

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: samples/managed-app-sample/mainTemplate.json [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+            "type": "Bool",
+            "metadata": {
+                "description": "Boolean flag to turn on and off of RBAC."
+            }
+        },
+        "enablePrivateCluster": {
+            "defaultValue": false,
+            "type": "Bool",
+            "metadata": {
+                "description": "Enable private network access to the Kubernetes cluster."
```
Comment (ShashiShekhar32) [inline samples/managed-app-sample/mainTemplate.json]: Nit: Boolean flag to turn on and off private network access to the Kubernetes cluster. ?

File: samples/managed-app-sample/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    "userAssignedIdentities": {
+        "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', concat(parameters('clusterResourceName'), '-identity'))]": {}
+    }
+
+},
+```
+
+## UIDefinition
+
+In this sample [createUiDefinition](./createUiDefinition.json), we have remove extra components that is not applicable for the managed app offer.
```
Comment (ShashiShekhar32) [inline samples/managed-app-sample/README.md]: nit: we have removed extra components that are not applicable for the managed app offer

File: samples/managed-app-sample/mainTemplate.json [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    "variables": {
+        "plan-name": "REPLACE-WITH-YOUR-PLAN-ID",
+        "plan-publisher": "REPLACE-WITH-YOUR-PUBLISHER-ID",
+        "plan-offerID": "REPLACE-WITH-YOUR-OFFER-ID",
+        "releaseTrain": "REPLACE-WITH-YOUR-RELEASE-TRAIN",
+        "clusterExtensionTypeName": "REPLACE-WITH-YOUR-EXTENSION-TYPE-NAME"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
```
Comment (ShashiShekhar32) [inline samples/managed-app-sample/mainTemplate.json:133]: curious, would CPA tool allow adding a managed identity ? I remember we only allow AKS cluster and extension resources creation

Comment (thomasyip-msft) [reply samples/managed-app-sample/mainTemplate.json:133]: no it doesn't. This mainTemplate and UiDefinition is for managed app.

File: README.md [hunk 4]
```diff
@@ -7,9 +7,13 @@ products:
 - azure-marketplace
 ---
 
-# Azure Marketplace Kubernetes App samples
+# Azure Marketplace Kubernetes App quick starting Guide
```
Comment (RamyasreeChakka) [inline README.md]: Please change the title to "Azure Marketplace Kubernetes Apps" #Closed

File: README.md [hunk 5]
```diff
@@ -7,9 +7,13 @@ products:
...
+
+## Getting started guide
+
+- [Getting Started: Publishing a Kubernetes App on Azure Marketplace](getting-started/GettingStarted.md)
+- [Microsoft Defender: Vulnerability Pre-Scan Check](getting-started/Vulnerability-Scan.md)
 
-This repository contains samples for creating Kubernetes Applications in Azure Marketplace.
```
Comment (RamyasreeChakka) [inline README.md:12]: Can you add text "This repository contains getting started guide and samples for creating Kubernetes Applications in Azure Marketplace."

File: samples/managed-app-sample/createUiDefinition.json [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+	"handler": "Microsoft.Azure.CreateUIDef",
+	"version": "0.1.2-preview",
+	"parameters": {
+		"config": {
+			"isWizard": false,
+			"basics": {
+				"location": {
+					"visible": "[basics('createNewCluster')]",
+					"allowedValues": [
+						"eastus",
```
Comment (RamyasreeChakka) [inline samples/managed-app-sample/createUiDefinition.json:12]: Currently we support a total of 33 regions. Can you please add all the 33 locations in all the CreateUIDef files? #Closed

File: samples/managed-app-sample/createUiDefinition.json [hunk 7]
```diff
@@ -0,0 +1,189 @@
+{
```
Comment (RamyasreeChakka) [inline samples/managed-app-sample/createUiDefinition.json:1]: Can you please add a README file in this sample folder and explain the concept? I am not able to understand easily. #Closed

File: samples/managed-app-sample/createUiDefinition.json [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+	"$schema": "https://schema.management.azure.com/schemas/0.1.2-preview/CreateUIDefinition.MultiVm.json#",
+	"handler": "Microsoft.Azure.CreateUIDef",
```
Comment (RamyasreeChakka) [inline samples/managed-app-sample/createUiDefinition.json:3]: To improve discoverability, can you please include this sample in the table in repo's readme.md https://github.com/Azure-Samples/kubernetes-offer-samples#samples-in-this-repo ? #Closed

File: samples/k8s-offer-azure-vote-custom-meters/createUIDefinition.json [hunk 9]
```diff
@@ -215,7 +215,7 @@
 						"defaultValue": "Cats",
 						"constraints": {
 							"required": true,
-							"regex": "^[a-z0-9A-Z]{1,30}$",
+							"regex": "^[a-z0-9A-Z ]{1,30}$",
```
Comment (RamyasreeChakka) [inline samples/k8s-offer-azure-vote-custom-meters/createUIDefinition.json:224]: Not related to your PR. I just noticed README is missing in k8s-offer-azure-vote and k8s-offer-azure-vote-custom-meters application. Can you please add a README in both these folders?  #Closed

File: samples/managed-app-sample/createUiDefinition.json (hunk 7 above)
Comment (thomasyip-msft) [reply samples/managed-app-sample/createUiDefinition.json:1]: Actually this was purposely taken out. We will add this back once Maanasa publishes hers to link to the learn microsoft doc.

**Review by RamyasreeChakka — APPROVED** [review]
:shipit:

**Review by RamyasreeChakka — APPROVED** [review]

**Pull request merged**